include/ddc/chunk_span.hpp

```diff
@@ -260,30 +262,58 @@ class ChunkSpan<ElementType, DiscreteDomain<DDims...>, LayoutStridedPolicy, Memo
         using OutDDom = detail::convert_type_seq_to_discrete_domain_t<OutTypeSeqDDims>;
-        return ChunkSpan<
-                ElementType,
-                OutDDom,
-                typename decltype(subview)::layout_type,
-                memory_space>(subview, OutDDom(this->m_domain));
+        if constexpr (
+                !std::is_same_v<layout_type, Kokkos::Experimental::layout_left_padded<>>
+                && !std::is_same_v<layout_type, Kokkos::Experimental::layout_right_padded<>>) {
+            return ChunkSpan<
+                    ElementType,
+                    OutDDom,
+                    layout_type,
+                    memory_space>(subview, OutDDom(this->m_domain));
+        } else {
+            Kokkos::layout_stride::mapping<extents_type> const mapping_stride(subview.mapping());
+            Kokkos::mdspan<ElementType, extents_type, Kokkos::layout_stride> const
+                    a(subview.data_handle(), mapping_stride);
+            return ChunkSpan<
+                    ElementType,
+                    OutDDom,
+                    Kokkos::layout_stride,
+                    memory_space>(a, OutDDom(this->m_domain));
+        }
     }
 
     /** Restrict to a subdomain
      */
     template <class... QueryDDims>
     KOKKOS_FUNCTION constexpr auto operator[](DiscreteDomain<QueryDDims...> const& odomain) const
     {
-        auto subview = std::experimental::
-                submdspan(allocation_mdspan(), get_slicer_for<DDims>(odomain)...);
-        return ChunkSpan<
-                ElementType,
-                decltype(this->m_domain.restrict(odomain)),
-                typename decltype(subview)::layout_type,
-                memory_space>(subview, this->m_domain.restrict(odomain));
+        auto subview = Kokkos::submdspan(allocation_mdspan(), get_slicer_for<DDims>(odomain)...);
+        using layout_type = typename decltype(subview)::layout_type;
+        using extents_type = typename decltype(subview)::extents_type;
+        if constexpr (
+                !std::is_same_v<layout_type, Kokkos::Experimental::layout_left_padded<>>
+                && !std::is_same_v<layout_type, Kokkos::Experimental::layout_right_padded<>>) {
+            return ChunkSpan<
+                    ElementType,
+                    decltype(this->m_domain.restrict(odomain)),
+                    layout_type,
+                    memory_space>(subview, this->m_domain.restrict(odomain));
+        } else {
+            Kokkos::layout_stride::mapping<extents_type> const mapping_stride(subview.mapping());
+            Kokkos::mdspan<ElementType, extents_type, Kokkos::layout_stride> const
+                    a(subview.data_handle(), mapping_stride);
+            return ChunkSpan<
+                    ElementType,
+                    decltype(this->m_domain.restrict(odomain)),
+                    Kokkos::layout_stride,
+                    memory_space>(a, this->m_domain.restrict(odomain));
+        }
```

Comment:
Same for previous remark

Reply:
fixed in 49c89244b2b6db06209098c6cf4c5a9676f8b5ae